lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -77,43 +77,35 @@ class WP_Theme_JSON_Resolver_Gutenberg {
 	 */
 	protected static $user_custom_post_type_id = null;
 
-	/**
-	 * Container to keep loaded i18n schema for `theme.json`.
-	 *
-	 * @since 5.8.0 As `$theme_json_i18n`.
-	 * @since 5.9.0 Renamed from `$theme_json_i18n` to `$i18n_schema`.
-	 * @var array
-	 */
-	protected static $i18n_schema = null;
-
-	/**
-	 * `theme.json` file cache.
-	 *
-	 * @since 6.1.0
-	 * @var array
-	 */
-	protected static $theme_json_file_cache = array();
-
 	/**
 	 * Processes a file that adheres to the theme.json schema
 	 * and returns an array with its contents, or a void array if none found.
 	 *
 	 * @since 5.8.0
 	 * @since 6.1.0 Added caching.
+	 * @since 6.5.0 Added persistent caching using object cache, and the `$cache_key` parameter.
 	 *
 	 * @param string $file_path Path to file. Empty if no file.
+	 * @param string $cache_key Optional. Key to cache the result under. Omitting the parameter results in no caching
+	 *                          being used. Default empty string.
 	 * @return array Contents that adhere to the theme.json schema.
 	 */
-	protected static function read_json_file( $file_path ) {
+	protected static function read_json_file( $file_path, $cache_key = '' ) {
 		if ( $file_path ) {
-			if ( array_key_exists( $file_path, static::$theme_json_file_cache ) ) {
-				return static::$theme_json_file_cache[ $file_path ];
+			$cache_group = 'theme_json_files';
```

Comment:
@joemcgill Related to https://core.trac.wordpress.org/ticket/59719#comment:18, what is your thought on this?

* The PR currently uses `theme_json_files`, yet another cache group name than what you're describing.
* Out of the 3 names we've been discussing in that ticket, this fits best into `theme_files`, as it's global data and the cache keys already include the theme slug. It's also exclusively about data from the filesystem (i.e. unlike the `theme_json` cache group), plus it needs to be persistent to be beneficial (i.e. unlike the `themes` cache group).
* One caveat is that currently this cache is also used for `theme.json` files that are not from a theme. Hence the `theme_json_files` cache group name makes sense here, whereas using a broader `theme_files` cache group name would be misleading in that sense, as core's `theme.json` is not really a theme file.

I think we have 3 options here:
1. Change this to become part of the `theme_files` cache group, and acknowledge that that cache group has a special case for WP core's own "fallback" theme files (specifically in this case, the `theme.json` file). In that case, the only other change needed here would be to adjust the cache key names so that they include a mention of `theme_json`, to clarify that these are cache keys for a `theme.json` file, rather than another file from a theme.
2. Keep this as is. Probably best from a separation of concerns and understandability perspective, but yet another cache group related to theme data may seem excessive.
3. Combine this with what we've so far considered under the `theme_files` cache group, but use another name that is broadly about _any_ files. In that case, the cache keys would not only contain the theme slug, but also some prefix like `theme_{themeSlug}`, so that it could also cover files that are in core, or in theory even plugins.